### ISSUE TITLE
[Redesign] IN modal

### ISSUE THREAD

**Issue by boogheta** [medialab/hyphe-browser#85]
2 usecases of this modal: 
- a) Setting to IN a webentity from the "Browsed webentity" sidebar
- b) Adding a new webentity from the + button in the urlbar

Todo:
- [x] integrate modal from mockup
- [x] anticipate that all messages (title, doc and step titles) can be different for the two usecases
- [x] isolate state of input data so that it doesn't impact the current entity if canceled
- [x] each step has a confirm button, all further steps are disabled until the confirm has been done, all confirms need to be done before the whole form can be validated
- [x] for each step user can either click on Confirm, or make a change. If confirm was already done and a new change is done, confirm needs to be clicked again for this step and following ones
- [x] Step 1 : 
   + if coming from usecase a) : **default prefix** is the longest current entity prefix matching the URL, **minimum allowed prefix** is the longest current entity prefix matching the URL
   + if coming from usecase b) : **default prefix** is the whole url, **minimum allowed prefix** is the prefix of the original webentity + 1 level
- [x] Step 1 (scope) : disable all blocks lower than the current one (for instance if entity is `com|facebook`, current page is `com|facebook|group|coco`, user can only slide on group and coco)
- [x] When attempting to validate step 1, check for WE with same prefix (ask @boogheta for the proper api calls to do for this) - display a loader, then an error message if a WE with same prefix already exists ()
  - Voir https://github.com/medialab/hyphe/blob/master/doc/api.md#define-webentities `get_lru_definedprefixes` pour tout les prefix et p-e `get_webentity_by_lruprefix`

- [x] When changing scope (step 1), the list of pages in step 2 (homepage) is filtered to match the choice of prefix (just by checking client-side the prefix, no need to query again the list of pages to the API), and the automatic proposed name (step 3) is updated following the TitleCase current rule 
- [x] Step 2 use CardsList and KnownPageCard components sorted by alphabetical urls (from the shortest to the longest)
- [x] Step 2: default proposed homepage is none
- [x] Step 3: default proposed name is set automatically by the TitleCase rule (cf step1)
- [x] Optional Step 4: if scope was changed, and current entity has tags and/or research notes, add a 4th step with a checkbox, checked by default, proposing to replicate these tags into the new entity 
- [x] validation button activates a waitloader
- [x] send input data via API requests only after validation button has two different cases:
  * if coming from usecase a) and scope wasn't changed (correspond to what happened previously when clicking the IN button):
    + call store.basic_edit_webentity API query and set status to IN, name to confirmed name, and homepage to confirmed homepage
    + call store.crawl_webentity
    + close modal
  * if coming from usecase a) with a change of scope, or coming from usecase b):
    + call store.get_webentity_by_lruprefix with proposed prefix from scope, then 3 cases:
      - If not entity existing with this prefix, create it by calling declare_webentity_by_lru with the prefix, the name, IN as status and lruVariations to true, then store.set_webentity_homepage
      - <s>If entity exists and has status DISCOVERED: then use it: call store.basic_edit_webentity API query and set status to IN, name to confirmed name, and homepage to confirmed homepage</s>
      - <s>If entity exists and has status IN, OUT, or UND, then display custom modal (#84) proposing either to cancel, or to navigate to this entity</s>
    + if Step 4 checked call store.add_webentity_tag_value on the new entity for each tag of the original entity 
    + call store.crawl_webentity on the resulting entity from previous step
    + close modal and refresh sidebar with the new entity
- [x] what happens to current entity when creating a new one? TBD (automatically OUT? stays DISOVERED?) --> after discussion: we do nothing !

- [x] when validating a step, autoscroll next step top border at center of modal

**Comment by boogheta** [comment]
The webentity created with the modal is not set as IN not crawled yet right?
